File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/financeiro/contabancaria/ContaBancariaGridDTO.java
package br.com.grupopipa.gestaointegrada.financeiro.contabancaria;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.GridDTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Builder
@Getter
@Setter
public class ContaBancariaGridDTO implements GridDTO {
    private UUID id;
    private String nome;
    private String banco;
    private String tipo;
    private BigDecimal saldoInicial;
    private String unidadeNegocioCodigo;
    private Boolean ativa;
    private List<String> formasPagamento;
    private Boolean deleted;
}

File: src/frontend/src/app/components/financeiro/conta-bancaria/model/conta-bancaria-grid-dto.ts
export class ContaBancariaGridDTO {
  id: string;
  nome: string;
  banco?: string;
  tipo: string;
  saldoInicial?: number;
  unidadeNegocioCodigo?: string;
  ativa: boolean;
  formasPagamento?: string[];
  deleted?: boolean;

  constructor(
    id: string,
    nome: string,
    banco: string | undefined,
    tipo: string,
    saldoInicial: number | undefined,
    unidadeNegocioCodigo: string | undefined,
    ativa: boolean,
    formasPagamento?: string[],
    deleted?: boolean
  ) {
    this.id = id;
    this.nome = nome;
    this.banco = banco;
    this.tipo = tipo;
    this.saldoInicial = saldoInicial;
    this.unidadeNegocioCodigo = unidadeNegocioCodigo;
    this.ativa = ativa;
    this.formasPagamento = formasPagamento;
    this.deleted = deleted;
  }
}
